cibyl/utils/arguments.py

```diff
@@ -0,0 +1,41 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+from typing import Dict, List
+
+from cibyl.cli.argument import Argument
+from cibyl.utils.dicts import subset
+
+
+def has_filtering_arguments(args: Dict[str, Argument],
+                            args_to_check: List[str]) -> bool:
+    """Check if the args dictionary contains any argument with values to
+    filter, specified in of the args_to_check list.
+
+    :returns: Whether there is any argument in args, of the ones specified in
+    the args_to_check list, that has a value to filter the output.
+    """
+    test_args = subset(args, args_to_check)
+    return any(arg.value for arg in test_args.values())
+
+
+def has_filter_tests(args: Dict[str, Argument]) -> bool:
+    """Check if the kwargs contain any argument with value to filter tests.
+
+    :returns: Whether there is any tests-related argument that has a value to
+    filter the found tests.
+    """
+    return has_filtering_arguments(args,
+                                   ["tests", "test_result", "test_duration"])
```

Comment:
I do not see this as easily expandable. If someone later on adds a new filter for tests on the build model, most probably they will not be aware of this and will forget to add it to the list. That is a bug waiting to happen.

Reply:
That's a good point, I created this function since the exact same call appears in Jenkins and Elasticsearch source, but I see how having the arguments listed in the utils subpackage might lead to problems, I'll move it back to the source.

Reply:
I'm reverting the changes to the Jenkins source and adopt slightly different solutions in the two sources for this problem. I'll revisit the Jenkins implementation in another change.